flink-tests/src/test/java/org/apache/flink/test/checkpointing/ResumeCheckpointManuallyITCase.java

```diff
@@ -288,26 +334,31 @@ private void testExternalizedCheckpoints(
 
     private static String runJobAndGetExternalizedCheckpoint(
             StateBackend backend,
-            File checkpointDir,
             @Nullable String externalCheckpoint,
-            ClusterClient<?> client)
+            MiniClusterWithClientResource cluster,
+            RestoreMode restoreMode)
             throws Exception {
-        JobGraph initialJobGraph = getJobGraph(backend, externalCheckpoint);
+        JobGraph initialJobGraph = getJobGraph(backend, externalCheckpoint, restoreMode);
         NotifyingInfiniteTupleSource.countDownLatch = new CountDownLatch(PARALLELISM);
-
-        client.submitJob(initialJobGraph).get();
+        cluster.getClusterClient().submitJob(initialJobGraph).get();
 
         // wait until all sources have been started
         NotifyingInfiniteTupleSource.countDownLatch.await();
 
-        TestUtils.waitUntilExternalizedCheckpointCreated(checkpointDir);
-        client.cancel(initialJobGraph.getJobID()).get();
-        TestUtils.waitUntilJobCanceled(initialJobGraph.getJobID(), client);
+        waitForCheckpoint(initialJobGraph.getJobID(), cluster.getMiniCluster(), 2);
```

Comment:
Add a comment to describe we need at least 2 checkpoints here.

Reply:
Added the comment.